docs/advanced.rst

```diff
@@ -505,3 +505,180 @@ Besides the stage directory, additional mount points can be specified through th
+	[ RUN      ] OSUAllreduceTest_8 on daint:gpu using PrgEnv-pgi
+	[      DEP ] OSUAllreduceTest_8 on daint:gpu using PrgEnv-pgi
+	[----------] finished processing OSUAllreduceTest_8 (OSU Allreduce test)
+
+	[----------] started processing OSUAllreduceTest_16 (OSU Allreduce test)
+	[ RUN      ] OSUAllreduceTest_16 on daint:gpu using PrgEnv-gnu
+	[      DEP ] OSUAllreduceTest_16 on daint:gpu using PrgEnv-gnu
+	[ RUN      ] OSUAllreduceTest_16 on daint:gpu using PrgEnv-intel
+	[      DEP ] OSUAllreduceTest_16 on daint:gpu using PrgEnv-intel
+	[ RUN      ] OSUAllreduceTest_16 on daint:gpu using PrgEnv-pgi
+	[      DEP ] OSUAllreduceTest_16 on daint:gpu using PrgEnv-pgi
+	[----------] finished processing OSUAllreduceTest_16 (OSU Allreduce test)
+
+	[----------] waiting for spawned checks to finish
+	[       OK ] OSUBuildTest on daint:gpu using PrgEnv-pgi
+	[       OK ] OSUBuildTest on daint:gpu using PrgEnv-gnu
+	[       OK ] OSUAllreduceTest_2 on daint:gpu using PrgEnv-pgi
+	[       OK ] OSUAllreduceTest_8 on daint:gpu using PrgEnv-gnu
+	[       OK ] OSUAllreduceTest_16 on daint:gpu using PrgEnv-gnu
+	[       OK ] OSUBuildTest on daint:gpu using PrgEnv-intel
+	[       OK ] OSULatencyTest on daint:gpu using PrgEnv-gnu
+	[       OK ] OSUBandwidthTest on daint:gpu using PrgEnv-gnu
+	[       OK ] OSUAllreduceTest_2 on daint:gpu using PrgEnv-gnu
+	[       OK ] OSUAllreduceTest_4 on daint:gpu using PrgEnv-pgi
+	[       OK ] OSUAllreduceTest_16 on daint:gpu using PrgEnv-pgi
+	[       OK ] OSULatencyTest on daint:gpu using PrgEnv-intel
+	[       OK ] OSUAllreduceTest_4 on daint:gpu using PrgEnv-intel
+	[       OK ] OSUAllreduceTest_16 on daint:gpu using PrgEnv-intel
+	[       OK ] OSUBandwidthTest on daint:gpu using PrgEnv-pgi
+	[       OK ] OSUAllreduceTest_8 on daint:gpu using PrgEnv-pgi
+	[       OK ] OSUAllreduceTest_8 on daint:gpu using PrgEnv-intel
+	[       OK ] OSUAllreduceTest_4 on daint:gpu using PrgEnv-gnu
+	[       OK ] OSULatencyTest on daint:gpu using PrgEnv-pgi
+	[       OK ] OSUAllreduceTest_2 on daint:gpu using PrgEnv-intel
+	[       OK ] OSUBandwidthTest on daint:gpu using PrgEnv-intel
+	[----------] all spawned checks have finished
+
+	[  PASSED  ] Ran 21 test case(s) from 7 check(s) (0 failure(s))
+	[==========] Finished on Tue Dec 10 00:21:11 2019
+
+Before starting running the tests, ReFrame topologically sorts them based on their dependencies and schedules them for running using the selected execution policy.
+With the serial execution policy, ReFrame simply executes the tests to completion as they "arrive", since the tests are already topologically sorted.
+In the asynchronous execution policy, tests are spawned and not waited for.
+If a test's dependencies have not yet completed, it will not start its execution and a ``DEP`` message will be printed to denote this.
+
+Finally, ReFrame's runtime takes care of properly cleaning up the resources of the tests respecting dependencies.
```

Comment:
```suggestion
Finally, ReFrame's runtime takes care of properly cleaning up the resources of the tests' respecting dependencies.
```

Reply:
I don't think the apostrophe here is correct.

Reply:
Ah I am sorry, I thought you meant `their respective dependencies`.